libbeat/processors/actions/replace.go

```diff
@@ -36,15 +35,15 @@ type replaceString struct {
 }
 
 type replaceStringConfig struct {
-	Fields        []replaceWith `config:"fields"`
+	Fields        []replaceConfig `config:"fields"`
 	IgnoreMissing bool          `config:"ignore_missing"`
 	FailOnError   bool          `config:"fail_on_error"`
 }
 
-type replaceWith struct {
-	FieldName   string `config:"fieldname"`
-	SearchValue string `config:"search_value"`
-	ReplaceWith string `config:"replace_with"`
+type replaceConfig struct {
+	Field   string `config:"field"`
+	Pattern string `config:"pattern"`
```

Comment:
@urso @exekias should https://github.com/elastic/beats/blob/master/libbeat/common/match/matchers.go be enhanced to do replacing strings and just use `matcher.Match` here?

Reply:
for Pattern in the config use `Patter *regexp.Regexp`. Config unpacking will automatically try to compile the regex and fail with error + setting name if this has failed.

`matcher.Match` is an optimization for some custom cases only, but falls back to `regexp` if the case becomes more 'complicated'. Given the optimizations we have in matcher, I only see the case for a constant string match being helpful (which would become a sub-string search, or in some cases string-prefix/suffix comparison).

The `matcher` package also replaces capturing-group-matches with non-capturing-groups (greatly reduces allocations). Having patterns and replacement like `gsub`, do we want to allow users to use capturing group in the replacement in the future? E.g.

```
pattern: 'some (?P<important>[a-zA-Z]) string'
replace: 'found: {{important}}'
```

For now I would not enhance the `matcher` package. Only if we figure this is indeed a common problem. When doing so we might have to remove some of the optimizations. In case we find we really need to optimize another type (e.g. matcher.Replacer) might give us better flexibility in applying the kind of optimizations we need for the use-case, while not un-optimizing `matcher.Matcher`.

Reply:
@urso Changed to pattern, as suggested.